dotnet/src/Extensions/TemplateEngine.PromptTemplateEngine/Blocks/NamedArgBlock.cs

```diff
@@ -0,0 +1,182 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Orchestration;
+
+namespace Microsoft.SemanticKernel.TemplateEngine.Prompt.Blocks;
+
+/// <summary>
+/// A <see cref="Block"/> that represents a named argument for a function call.
+/// For example, in the template {{ MySkill.MyFunction var1="foo" }}, var1="foo" is a named arg block.
+/// </summary>
+internal sealed class NamedArgBlock : Block, ITextRendering
+{
+    /// <summary>
+    /// Returns the <see cref="BlockTypes"/>.
+    /// </summary>
+    internal override BlockTypes Type => BlockTypes.NamedArg;
+
+    /// <summary>
+    /// Gets the name of the function argument.
+    /// </summary>
+    internal string Name { get; } = string.Empty;
+
+    /// <summary>
+    /// Gets the rendered value of the function argument. If the value is a <see cref="ValBlock"/>, the value stays the same.
+    /// If the value is a <see cref="VarBlock"/>, the value of the variable is determined by the context variables passed in.
+    /// </summary>
+    /// <param name="variables">Variables to use for rendering the named argument value when the value is a <see cref="VarBlock"/>.</param>
+    /// <returns></returns>
+    internal string GetValue(ContextVariables? variables)
+    {
+        var valueIsValidValBlock = this._valBlock != null && this._valBlock.IsValid(out var errorMessage);
+        if (valueIsValidValBlock)
+        {
+            return this._valBlock!.Render(variables);
+        }
+
+        var valueIsValidVarBlock = this._argValueAsVarBlock != null && this._argValueAsVarBlock.IsValid(out var errorMessage2);
+        if (valueIsValidVarBlock)
+        {
+            return this._argValueAsVarBlock!.Render(variables);
+        }
+
+        return string.Empty;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="NamedArgBlock"/> class.
+    /// </summary>
+    /// <param name="text">Raw text parsed from the prompt template.</param>
+    /// <param name="logger">The <see cref="ILoggerFactory"/> to use for logging. If null, no logging will be performed.</param>
+    /// <exception cref="SKException"></exception>
+    public NamedArgBlock(string? text, ILoggerFactory? logger = null)
```

Comment:
I would place constructor above `GetValue` method and remove `return` statement, since it's not required in constructor. Ideally, it would be great to move the logic with exception throwing and logging to separate methods to keep constructor clean and simple.

Reply:
Makes sense. This started out as a property and never got moved. And good catch on the `return`.